v31_migration_guide.md

```diff
@@ -0,0 +1,135 @@
+like this:
+
+``` go
+type UsageRecordParams struct {
+	Quantity uint64 `form:"quantity"`
+}
+```
+
+We were unable to differentiate these two cases:
+
+``` go
+// Initialized with no quantity
+UsageRecord {}
+
+// Initialized with an explicitly zero
+UsageRecord {
+    Quantity: 0,
+}
+```
+
+This is because any uninitialized fields on a struct in Go are set to their
+type's "zero value", which for an integer is `0`.
+
+Working around the problem required a secondary field to help explicitly state
+that the zero value was intended, which was quite unintuitive for users:
+
+``` go
+type UsageRecordParams struct {
+	Quantity     uint64 `form:"quantity"`
+	QuantityZero bool   `form:"quantity,zero"`
+}
+
+UsageRecord {
+    QuantityZero: true,
+}
+```
+
+To address the problem, we moved every parameter structure over to use pointers
+instead. So the above becomes:
+
+``` go
+type UsageRecordParams struct {
+	Quantity *int64 `form:"quantity"`
+}
+```
+
+Because in Go you can't take the address of an inline value (`&0`), we provide
+a set of helper functions like `stripe.Int64` specifically for initializing
+these structs:
+
+``` go
+UsageRecord {
+    Quantity: string.Int64(0),
+}
+```
+
+The zero value for pointers is `nil`, so we can now easily determine which
+values on a struct were never set, and which ones were explicitly set to a zero
+value, thus eliminating the need for the secondary fields like `QuantityZero`.
+
+Because this is a large change, we also took the opportunity to do some
+housekeeping throughout the library. Most of this involves renaming fields and
+some resources to be more accurate according to the API, but it also involves
+some smaller changes like moving some types and constants around. Please see
+the list below for the complete set of changes.
+
+## Major changes
+
+* All fields on parameter structs (those that end with `*Params`) are now
+  pointers. Please use the new helper functions to set them:
+    * `stripe.Bool`
+    * `stripe.Float64`
+    * `stripe.Int64`
+    * `stripe.String`
+    This also means that extra fields that used to be solely used for tracking
+    meaningful zero values like `CouponEmpty` and `QuantityZero` have been
+    dropped. Use their corresponding field (e.g., `Coupon`, `Quantity`) with an
+    explicit empty value instead (`stripe.String("")`, `stripe.Int64(0)`).
+* Many fields have been renamed so that they're more consistent with their name
+  in Stripe's REST API. Most of the time, this changes abbreviations to a more
+  fully expanded form. For example:
+    * `Desc` becomes `Description`.
+    * `Live` becomes `Livemode`.
+* A few names of API resources (and their corresponding parameter and list
+  classes) have changed:
+    * `Fee` becomes `ApplicationFee`.
+    * `FeeRefund` becomes `ApplicationFeeRefund`.
+    * `Sub` becomes `Subscription`.
+    * `SubItem` becomes `SubscriptionItem`.
+    * `Transaction` becomes `BalanceTransaction`.
```

Comment:
Did I miss any here?

Reply:
`Owner` is now `AdditionalOwner` though it's not really an API resource so it might be fine to ignore